### DISCUSSION TITLE
ext-auth external call

### DISCUSSION THREAD

**Discussion by RazaGR** [strongswan/strongswan#2394] [Q&A]
can the [ext-auth plugin](https://docs.strongswan.org/docs/5.9/plugins/ext-auth.html) receive EAP username and password  as environment variables which user provided during login?

currently those are stored in `/etc/ipsec.secrets `

I have read about eap-radius but I would like to see possibility with just calling my external http server which has users details, if possible to just send a POST request with username/pass

is there another plugin available for it? 
Thanks

**Comment by RazaGR** [comment]
I also setup [eap-radius in golang](https://github.com/layeh/radius) but its not passing  attribute `password` to my server, I get only username [as per documentation which attributes are passed ](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_attributes_sent_to_radius_servers)

I notice [XAuth](https://docs.strongswan.org/docs/5.9/plugins/eap-radius.html#_xauth_via_radius) pass that password but I am using ikev2  `keyexchange=ikev2...rightauth=eap-radius`

if I can somehow pass the password to my radius server it will help or is there any other possibility using auth with ikev2

**Reply by tobiasbrunner** [reply]
EAP methods don't generally transport cleartext passwords. The only method that is implemented by strongSwan and does so is [EAP-GTC](https://docs.strongswan.org/docs/5.9/plugins/eap-gtc.html), however, it's not widely supported by other implementations.
